Source/Core/InputCommon/GCAdapter.cpp

```diff
@@ -83,6 +88,22 @@ static void Read()
   }
 }
 
+static void Write()
+{
+  int size = 0;
+  while (s_adapter_thread_running.IsSet())
+  {
+    if (s_rumble_data_available.WaitFor(std::chrono::milliseconds(100)))
```

Comment:
Yes. The standard approach is to just call Wait(), trigger s_rumble_data_available on shutdown, check for s_adapter_thread_running again and return immediately if it's false, so that shutdowns are never delayed by up to 100ms.